vision/src/autogluon/vision/image_classification/image_classification.py

```diff
@@ -1,341 +1,256 @@
-import logging
-import os
+"""Image classification task"""
 import copy
-import mxnet as mx
-
-from .pipeline import train_image_classification
-from autogluon.core.scheduler import get_cpu_count, get_gpu_count
-from autogluon.core.task.base import BaseTask, compile_scheduler_options, create_scheduler
-from autogluon.mxnet.task.classifier import Classifier
-from autogluon.mxnet.task.dataset import get_dataset
-from autogluon.mxnet.task.nets import *
-from autogluon.mxnet.task.utils import *
-from autogluon.mxnet.loss import SoftmaxCrossEntropyLoss
-from autogluon.mxnet.optimizer import NAG
-from autogluon.mxnet.utils import update_params
-import autogluon.core as ag
+import pickle
+import logging
+
+import pandas as pd
+from gluoncv.auto.tasks import ImageClassification as _ImageClassification
+from gluoncv.model_zoo import get_model_list
 
 __all__ = ['ImageClassification']
 
-from autogluon.core import sample_config
+class ImageClassification(object):
```

Comment:
`ImageClassifier`? `ImagePredictor`? `ImageClassification` doesn't seem like the right name for a class. I personally think `ImagePredictor` is best as it leaves the door open to expand to regression, and it will align with `TabularPredictor`.